### PR TITLE
startup asyn

### PR DESCRIPTION
**Pull request by chylli-binary**

### CODE REVIEW

File: tests/test_deriv_api.py [hunk 1]
```diff
@@ -1,25 +1,15 @@
...
 import json
+import pytest
 
-deriv_api_obj = deriv_api.deriv_api(1089)
 
-#def test_connection():
-#    json_data = json.dumps({'ping': 1})
-#    deriv_api_obj.wsconnection.send(json_data)
-#    deriv_api_obj.wsconnection.recv() == 'test'
+deriv_api_obj = deriv_api.DerivAPI(1089)
```
Comment (chylli-binary) [inline tests/test_deriv_api.py:7]: We shouldn't real connect to a live server. We should mock the connection function.

File: tests/test_deriv_api.py [hunk 2]
```diff
@@ -1,25 +1,15 @@
...
-    expected = "pong"
-    message = '{"echo_req": {"ping": 1}, "msg_type": "ping", "ping": "pong"}'
-    assert deriv_api_obj.on_pong(deriv_api_obj.wsconnection, message) == expected
-
-def test_on_message():
-    message = '{"echo_req": {"ping": 1}, "msg_type": "ping", "ping": "pong"}'
-    data = deriv_api_obj.on_message(deriv_api_obj.wsconnection, message)
-    assert data['msg_type'] == "ping"
-    assert data['ping'] == "pong"
+@pytest.mark.asyncio
```
Comment (chylli-binary) [inline tests/test_deriv_api.py:12]: want to know, what's the meaning of syntax `@` here ? can you give me the python doc ?

File: deriv_api/deriv_api.py [hunk 3]
```diff
@@ -14,38 +24,42 @@ def __init__(self, app_id, endpoint = 'frontend.binaryws.com', lang = 'EN', bran
...
+        if (self.wsconnection):
+            return self.wsconnection
+        else:
+            self.wsconnection = websockets.connect(self.__get_apiURL)
+            return 
 
-        return data
+    # this is called from client by api.buy, api.account... then in APICalls -> processRequest -> validateArgs + call this send
+    async def send(self, message):
+        loop.create_task(self.send_receive(message))
```
Comment (chylli-binary) [inline deriv_api/deriv_api.py:45]: Just want to know, why we need `create_task` here ? can't we use `await` directly ? what's the difference between `create_task` and `await` ?

Comment (chylli-binary) [reply deriv_api/deriv_api.py:45]: when should we `create_task` ? can't we `create_task(self.send` in the upper func ?

Comment (sasikala-binary) [reply deriv_api/deriv_api.py:45]: yes we can directly use await. If we need we can use directly in upper function in APICalls

Comment (chylli-binary) [reply deriv_api/deriv_api.py:45]: Sorry I'm not disapprove you. I just want to learn python.

File: deriv_api/deriv_api.py [hunk 4]
```diff
@@ -14,38 +24,42 @@ def __init__(self, app_id, endpoint = 'frontend.binaryws.com', lang = 'EN', bran
...
-    #    self.endpoint = endpoint
+    #def add_api_call(self, request):
+    #    loop.create_task(self.send_receive(request))
 
-    #def get_endpoint(self):
-    #    return self.endpoint
+loop = asyncio.get_event_loop()
+# connect & ping parallely
+#loop.create_task(apiconnect())
+loop.run_forever()
```
Comment (chylli-binary) [inline deriv_api/deriv_api.py:65]: I don't think we should add this line in the module. it should be in the entry script.

**Pull request merged**